sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1660,4 +1669,40 @@ static uint32_t _handle_rio(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
 
     return route_ltime;
 }
+
+static void _handle_timestamp(const ndp_opt_timestamp_t *tsmp)
+{
+    if (!IS_USED(MODULE_GNRC_IPV6_NIB_TIMESTAMP)) {
+        return;
+    }
+
+    uint64_t now = byteorder_ntohll(tsmp->timestamp);
+    DEBUG("nib: received Timestamp option:\n");
+    DEBUG("     - Seconds: %u\n", (unsigned)(now >> 16));
+    DEBUG("     - Milliseconds: %u\n", (unsigned)(1000 * (now & 0xFFFF)) >> 16);
+#if IS_USED(MODULE_RTT64)
+    static uint16_t diff;
+    uint64_t local = rtt64_get_counter();
+    now += diff;
+
+    rtt64_set_counter(now);
```

Comment:
This is not what the timestamp option is for. It is intended to detect replay attacks in unsolicited RA (see https://www.rfc-editor.org/rfc/rfc3971.html#section-5.3.1). Using it to synchronize the local time somewhat contradicts the standardized usage.